source/Analysis/AnnualStatistics.vb

```diff
@@ -135,6 +135,38 @@ Friend Class AnnualStatistics
                 stat.avg.ToString(formatstring),
                 stat.vol.ToString(formatstring)) & eol
         Next
+
+        'Generate timeseries with max/avg/min values if years are present
+        If (Me.stats.Count > 1) Then
+            'Prepare output timeseries
+            Dim basename As String = Me.InputTimeSeries(0).Title
+            Dim timeseries_max As TimeSeries = New TimeSeries(basename + "_max")
+            Dim timeseries_avg As TimeSeries = New TimeSeries(basename + "_avg")
+            Dim timeseries_min As TimeSeries = New TimeSeries(basename + "_min")
+
+            'Set interpretation mode
+            timeseries_max.Interpretation = TimeSeries.InterpretationEnum.BlockRight
+            timeseries_avg.Interpretation = TimeSeries.InterpretationEnum.BlockRight
+            timeseries_min.Interpretation = TimeSeries.InterpretationEnum.BlockRight
+
+            'Get Unit from input timeseries
+            Dim unit As String = Me.InputTimeSeries(0).Unit
+            timeseries_max.Unit = unit
+            timeseries_avg.Unit = unit
+            timeseries_min.Unit = unit
+
+            'Fill timeseries with values of max, avg, min
+            For Each kvp As KeyValuePair(Of String, struct_stat) In Me.stats

```

Comment:
`Me.stats` also contains an entry for the "Entire series" besides the annual values. This entry should probably not be added to the timeseries. I suggest either filtering `Me.stats` by key or skipping the first entry.